src/TrancheWrapper.sol

```diff
@@ -0,0 +1,236 @@
+/**
+ * Created by Pragma Labs
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity 0.8.22;
+
+import { ERC4626 } from "../lib/solmate/src/mixins/ERC4626.sol";
+import { ERC20 } from "../lib/solmate/src/tokens/ERC20.sol";
+import { FixedPointMathLib } from "../lib/solmate/src/utils/FixedPointMathLib.sol";
+import { ITranche } from "./interfaces/ITranche.sol";
+import { ILendingPool } from "./interfaces/ILendingPool.sol";
+import { SafeTransferLib } from "../lib/solmate/src/utils/SafeTransferLib.sol";
+
+/**
+ * @title Tranche Wrapper
+ * @notice This contract wraps a Tranche to make it ERC4626 compliant.
+ * @dev This contract acts as a wrapper for interactions with the Tranche, ensuring ERC4626-compliance.
+ * - Shares minted via the wrapper have a 1:1 ratio to shares minted on the Tranche.
+ * - All view functions can be called directly on the Tranche view functions.
+ * - Deposit and Mint functions approve the LendingPool, not the Tranche.
+ */
+contract TrancheWrapper is ERC4626 {
+    using FixedPointMathLib for uint256;
+    using SafeTransferLib for ERC20;
+
+    /* //////////////////////////////////////////////////////////////
+                                CONSTANTS
+    ////////////////////////////////////////////////////////////// */
+
+    // Address of the Lending Pool.
+    address public immutable LENDING_POOL;
+    // Address of the underlying Tranche.
+    address public immutable TRANCHE;
+
+    /* //////////////////////////////////////////////////////////////
+                              CONSTRUCTOR
+    ////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice The constructor for a TrancheWrapper.
+     * @param asset_ The underlying ERC20 asset.
+     * @param name_ The name of the ERC4626 token.
+     * @param symbol_ The symbol of the ERC4626 token.
+     * @param TRANCHE_ The Tranche contract address.
+     */
+    constructor(ERC20 asset_, string memory name_, string memory symbol_, address TRANCHE_)
```

Comment:
Here we can actually call` asset`, `name` and `symbol` on the tranche itself, so no need to pass it to the constructor.